lib/Onyx.js

```diff
@@ -712,12 +713,12 @@ function clear() {
     return getAllKeys()
         .then((keys) => {
             _.each(keys, (key) => {
-                keyChanged(key, null);
-                cache.set(key, null);
+                const resetValue = lodashGet(defaultKeyStates, key, null);
+                keyChanged(key, resetValue);
+                cache.set(key, resetValue);
             });
         })
-        .then(Storage.clear)
-        .then(initializeWithDefaultKeyStates);
+        .then(Storage.clear);
```

Comment:
Looks good so far. I still have questions about what the effects of both `getAllKeys()` and `Storage.clear` would be and would like to see automated tests for the following cases:

**Case 1:**
```js
Onyx.clear();
Onyx.merge('someDefaultKey', 'someValue');

// Expect the value of the merge to replace the default key state
// Expect that storage matches what we have in the cache and that our call to 
// `Storage.clear()` did not clear the value of merge from storage after we called Onyx.clear()
```

**Case 2:**
```js
Onyx.clear();
Onyx.set('someDefaultKey', 'someValue');

// Expect the value of the set to replace the default key state
// Expect that storage matches what we have in the cache and that our call 
// to `Storage.clear()` did not prevent any values from being set
```

**Case 3:**
```js
Onyx.set('someDefaultKey', 'someValue');
Onyx.clear();

// Expect the value from the `set()` to be replaced with the default key state value
// Expect that storage matches what we have in the cache exactly
```

**Case 4:**
```js
Onyx.merge('someDefaultKey', 'someValue');
Onyx.clear();

// Expect the value from the `merge()` to be replaced with the default key state value
// Expect that storage matches what we have in the cache exactly
```

Reply:
These are great thanks, I will create some tests.